torch/csrc/api/include/torch/optim/serialize.h

```diff
@@ -207,27 +207,68 @@ void serialize(
   }
 }
 
+template <typename T>
+c10::List<T> deque_to_list(const std::deque<T>& dq) {
+  c10::List<Tensor> list;
+  list.reserve(dq.size());
+  for (const auto& e : dq) {
+    list.push_back(e);
+  }
+  return list;
+}
+
+template <typename T>
+void list_to_deque(const c10::List<T>& list, std::deque<T>& dq) {
+  for (size_t i = 0; i < list.size(); i++) {
+    dq.push_back(list[i]);
+  }
+}
+
 #define _TORCH_OPTIM_SERIALIZE(name) \
   torch::optim::serialize(archive, #name, self.name)
 
 #define _TORCH_OPTIM_SERIALIZE_WITH_TEMPLATE_ARG(OptimizerName) \
   torch::optim::serialize<OptimizerName##ParamState, OptimizerName##Options>(archive, self)
 
-#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG(name) \
-{ \
+#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG(name) { \
   auto ivalue = torch::IValue(name()); \
   /* do not serialize if name is an undefined tensor*/ \
   if (!(ivalue.isTensor() && ivalue.unsafeToTensorImpl() == at::UndefinedTensorImpl::singleton())) { \
     archive.write(#name, ivalue); \
   } \
 }
 
-#define _TORCH_OPTIM_DESERIALIZE_TORCH_ARG(T, name) \
-{ \
+#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG_DEQUE(name) { \
+  c10::IValue ivalue = torch::IValue(deque_to_list(name())); \
+  archive.write(#name, ivalue); \
+}
+
+#define _TORCH_OPTIM_DESERIALIZE_TORCH_ARG(T, name) { \
   c10::IValue ivalue; \
   bool exists = archive.try_read(#name, ivalue); \
   if (exists) \
     name(ivalue.to<T>()); \
 }
+
+#define _TORCH_OPTIM_DESERIALIZE_TORCH_ARG_OPTIONAL(T, name) { \
+  c10::IValue ivalue; \
+  archive.read(#name, ivalue); \
+  /*c10::nullopt is serialized which is consistent with the
+   Python API as it serializes None values. currently only options (and not state)
+   consist of c10::optional entries so we don't end up serializing anything that's not
+   serialized by Python API*/ \
```

Comment:
If we added the above mentioned comment to optimizer.h, then I think we can remove the comment here, so that there is only place where we need to maintain the correctness of this information :D